gems/aws-sdk-core/aws-sdk-core.gemspec

```diff
@@ -17,6 +17,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency('aws-partitions', '~> 1', '>= 1.239.0') # necessary for STS & S3 regional
   spec.add_dependency('aws-sigv4', '~> 1.1') # necessary for making Aws::STS API calls
   spec.add_dependency('aws-eventstream', '~> 1', '>= 1.0.2') # necessary for binary eventstream
+  spec.add_dependency('rexml', '>= 0') 
```

Comment:
Talked offline about this - I think instead, customers should depend on any XML parser that we support (bring your own parser), and we should not force dependencies that may not be necessary and to avoid potential breakage. In general we try to own and control our own dependencies.

Reply:
I agree with this.  The consequence however is that customers will need to specify (directly or indirectly) an xml library in order to use the SDK. In many cases rexml or another library will be brought in by other libraries anyway so this may not impact many users.  However, I will add descriptive error messages - currently we assume that there will always be at least one xml engine available and dont handle the missing case (since rexml was previously always available).

Reply:
Is there potential for reversing this decision? It seems odd to have an undeclared dependency, that wouldn't be discovered until runtime, and only when using Ruby 3+. 

Given that Ruby 3 still includes `rexml` as a bundled gem, it should generally be available everywhere anyhow.

IMO this is a breaking change that is a bit odd for minor version bump.

Reply:
Adding it as an explicit dependency breaks older rubies, which we must still support.  It's also not a required dependency: we support a number of xml engines, and will use whichever a client has loaded.    I don't believe that not including a new dependency is a breaking change for existing users - it might, in a few cases, require changes in a users code when they are upgrading from an older ruby version to 3.